### PR TITLE
[clang-tidy] fix misc-const-correctness to work with function-try-blocks

### LINKED ISSUE #99860: [clang-tidy] false negative for misc-const-correctness in function-try-block
The check `misc-const-correctness` of clang-tidy 19.0.0git (20d38158ffbe1e01b74e39d36d4b3bc5e7a63d66) does not work as expected for function-try-blocks.

For the following function the warning warning `variable 'i' of type 'int' can be declared 'const' [misc-const-correctness]` is generated.
```
int tp() {
  try {
    int i = 0;
    return i;
  } catch (...) {
    return 0;
  }
}
```
When the outer curly braces are ommited, the check does not report a warning.
```
int fn() try {
  int i = 0;
  return i;
} catch (...) {
  return 0;
}
```

### PR DESCRIPTION
**Pull request by schenker**
Make the clang-tidy check misc-const-correctness work with function-try-blocks.

Fixes #99860.

### CODE REVIEW

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -120,6 +120,9 @@ Improvements to clang-tidy
 - Improved :program:`check_clang_tidy.py` script. Added argument `-export-fixes`
   to aid in clang-tidy and test development.
 
+- Improved :doc:`misc-const-correctness
```
Comment (PiotrZSL) [inline clang-tools-extra/docs/ReleaseNotes.rst:123]: wrong place, this should be in line 375

**Review by PiotrZSL — CHANGES_REQUESTED** [review]
Release notes need small fix, except that looks fine

File: clang-tools-extra/docs/ReleaseNotes.rst (hunk 1 above)
Comment (schenker) [reply clang-tools-extra/docs/ReleaseNotes.rst]: done

**Review by PiotrZSL — APPROVED** [review]
LGTM

**Review by 5chmidti — APPROVED** [review]

**Comment by PiotrZSL** [comment]
Looks like missing -fexceptions

**Comment by schenker** [comment]
> Looks like missing -fexceptions

thanks for fixing it

**Pull request merged**